doozerlib/cli/release_gen_payload.py

```diff
@@ -504,6 +506,16 @@ def update_single_arch_istags(apiobj: oc.APIObject):
                     if adding_tags:
                         logger.warning(f'The following tag names are net new to {imagestream_namespace}:{imagestream_name}: {adding_tags}')
 
+            if publish:
+                arch_suffix = go_suffix_for_arch(arch)
+                priv_suffix = "-priv" if private_mode else ""
+                imagestream_name = f"{base_imagestream_name}{arch_suffix}{priv_suffix}"
+                imagestream_namespace = f"release{arch_suffix}{priv_suffix}"
+                publish_name = f"{runtime.get_minor_version()}.0-{runtime.assembly}"
+                publish_reponame = imagestream_namespace.replace("ocp", "release")
```

Comment:
This seems to be dangerous to do a string replacement because you are making an assumption that there is one but only one ocp substring in the imagestream_namespace. If a different namespace is given by the parameter, this will probably overwrite things unexpectedly. I believe to be explicitly configured either by command line arguments or config files.